src/Common/EncodingCharacterBase.ts

```diff
@@ -0,0 +1,74 @@
+import EncodingCategory from '../Common/EncodingCategory';
+import EncodingEntry from '../Common/EncodingEntry';
+import EncodingLookupResult from '../Common/EncodingLookupResult';
+import EncodingDataBase from './EncodingDataBase';
+
+abstract class EncodingCharacterBase<T extends number> {
```

Comment:
Why does this need to extend from number?

Reply:
This comes from EncodingDataBase<T> which requires that T extend from number.  Enums do extend from number, but if we wanted to have a more complicated encoding then we would need further refactoring.